crates/number/src/lib.rs

```diff
@@ -1,3 +1,4 @@
 pub mod conversions;
 pub mod nonzero;
 pub mod u256_decimal;
+pub mod u256_hex_or_decimal;
```

Comment:
nit: If we want all serialization to be hex or decimal, can we remove `u256_decimal` entirely?

Also, mostly a module naming nit, but can we

```suggestion
pub mod serialization;
```

And have `number::serialization::HexOrDecimalU256` (and, if we want to keep `u256_decimal` around, then `number::serialization::DecimalU256` as well).

I.e. I think all U256 serialization helpers should go in one module.